### ISSUE TITLE
Immutable se to true makes form return an error at every modelChange event

### ISSUE THREAD

**Issue by snekku-dev** [ngx-formly/ngx-formly#3584]
**Description**
After migrating from v5 to v6, with formly set with extras.immutable to true I get an error in console whenever the form emits a modelChange. I don't see anything related in the migration guide, or maybe is not that obvious to me unfortunately.

`Error: can't convert undefined to object`

**Minimal Reproduction**
The only changes from the base project were adding the immutable property to true and setting the model property in the formly-form component in the template as a "banana in the box".

Open the console and edit any field
https://stackblitz.com/edit/ngx-formly-ui-bootstrap-86bn3n?file=src/app/app.component.html

https://ngx-formly-ui-bootstrap-86bn3n.stackblitz.io

**Your Environment**

- Angular version: 14 (latest v14 stable)
- Formly version: 6 (latest stable)

**Comment by aitboudad** [comment]
The issue is caused by usage of two-way binding, rely on `modelChange` instead to get the new model value:

```patch
- [(model)]="model"
+ [model]="model"
+ (modelChange)="onModelChange($event)"
```

**Comment by snekku-dev** [comment]
> The issue is caused by usage of two-way binding, rely on `modelChange` instead to get the new model value:
> 
> ```diff
> - [(model)]="model"
> + [model]="model"
> + (modelChange)="onModelChange($event)"
> ```

I tried to do this in the example but I do still get the error

https://stackblitz.com/edit/ngx-formly-ui-bootstrap-86bn3n?file=src%2Fapp%2Fapp.component.html%2Csrc%2Fapp%2Fapp.component.ts

**Comment by aitboudad** [comment]
will check further, for a workaround you may separate initial model from the updated model:

```patch
- [model]="model"
+ [model]="initialModel"
```

**Comment by sokolej79** [comment]
We have same problem when set immutable: true. When we migrated from v5 to v 6.1.1 error is thrown: Cannot convert undefined or null to object.

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.1.2](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.1.2) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
